### PR TITLE
Add orm tests and split load methods

### PR DESCRIPTION
**Pull request by jamyouss**
- split loader methods
- add orm tests
- add `doctrine/orm` in composer depedencies

### CODE REVIEW

File: src/Majora/Component/OAuth/Tests/Loader/ORM/AccessTokenLoaderTest.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            DoctrineTestHelper::createTestEntityManager(),
+            new ClassMetadata(AccessTokenRepository::class)
+        );
+
+        $this->tokenLoader = new AccessTokenLoader(
+            $this->tokenRepository,
+            $this->clock
+        );
+    }
+}
```
Comment (Oliboy50) [inline src/Majora/Component/OAuth/Tests/Loader/ORM/AccessTokenLoaderTest.php]: EOF

File: src/Majora/Component/OAuth/Tests/Loader/ORM/AccountLoaderTest.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        $application = $this->prophesize('Majora\Component\OAuth\Entity\Application');
+
+        $queryBuilder = $accountLoader->retrieveOnApplicationByUsernameQueryBuilder($application->reveal(), Argument::any());
+
+        $query = $queryBuilder->getQuery();
+
+        $this->assertEquals($query->getDQL(), sprintf('SELECT ac FROM %s ac INNER JOIN ac.applications app WHERE ac.username = :username AND app.id = :application', AccountRepository::class));
+    }
+}
```
Comment (Oliboy50) [inline src/Majora/Component/OAuth/Tests/Loader/ORM/AccountLoaderTest.php]: EOF

File: src/Majora/Component/OAuth/Tests/Loader/ORM/ApplicationLoaderTest.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            $applicationRepository
+        );
+
+        $queryBuilder = $applicationLoader->retrieveByApiKeyAndSecretQueryBuilder(Argument::any(), Argument::any());
+
+        $query = $queryBuilder->getQuery();
+
+        $this->assertEquals($query->getDQL(), sprintf('SELECT a FROM %s a WHERE a.secret = :secret AND a.apiKey = :api_key', ApplicationRepository::class));
+    }
+}
```
Comment (Oliboy50) [inline src/Majora/Component/OAuth/Tests/Loader/ORM/ApplicationLoaderTest.php]: EOF

File: src/Majora/Component/OAuth/Tests/Loader/ORM/RefreshTokenLoaderTest.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            DoctrineTestHelper::createTestEntityManager(),
+            new ClassMetadata(RefreshTokenRepository::class)
+        );
+
+        $this->tokenLoader = new RefreshTokenLoader(
+            $this->tokenRepository,
+            $this->clock
+        );
+    }
+}
```
Comment (Oliboy50) [inline src/Majora/Component/OAuth/Tests/Loader/ORM/RefreshTokenLoaderTest.php]: EOF

File: src/Majora/Component/OAuth/Tests/Loader/ORM/TokenLoaderTest.php [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+     */
+    public function testRetrieveExpiredQueryBuilder()
+    {
+        $queryBuilder = $this->tokenLoader->retrieveExpiredQueryBuilder($this->clock->now());
+
+        $query = $queryBuilder->getQuery();
+
+        $this->assertEquals($query->getDQL(), sprintf('SELECT t FROM %s t WHERE t.expireAt < :date', $this->tokenRepository->getClassName()));
+    }
+}
```
Comment (Oliboy50) [inline src/Majora/Component/OAuth/Tests/Loader/ORM/TokenLoaderTest.php]: EOF

File: src/Majora/Component/OAuth/Tests/Loader/ORM/AccessTokenLoaderTest.php (hunk 1 above)
Comment (jamyouss) [reply src/Majora/Component/OAuth/Tests/Loader/ORM/AccessTokenLoaderTest.php]: Done

File: src/Majora/Component/OAuth/Tests/Loader/ORM/RefreshTokenLoaderTest.php (hunk 4 above)
Comment (jamyouss) [reply src/Majora/Component/OAuth/Tests/Loader/ORM/RefreshTokenLoaderTest.php]: Done

**Comment by jamyouss** [comment]
@Oliboy50 pourquoi les EOF ?

**Comment by jamyouss** [comment]
EOF fixed !

**Pull request merged**